### ISSUE TITLE
RFC: Add Introduction to Calculus from University of Sydney in Math Electives

### ISSUE THREAD

**Issue by daniel65dj** [ossu/computer-science#689]
MIT's Calculus 1A requires pre-calculus as a prerequisite and there are listed resources for studying pre-calculus. I would like to suggest the [Introduction to Calculus from University of Sydney](https://www.coursera.org/learn/introduction-to-calculus) as an option for studying those prerequisites because ossu students can probably use a refresher on precalculus and then transition smoothly to calculus. Also it's one of the top courses on coursera and full interactivity is free except for the certificate.

**Comment by Alaharon123** [comment]
It should be noted that Introduction to Calculus, besides covering the mentioned PreCalculus, covers the content of Calculus 1A and part of Calculus 1B. And has free quizzes you can skip to if you think you already know the content of any given section. It would seem to me to be a replacement for Calculus 1A as a more flexible course rather than a prerequisite although I have only looked at the course, not actually taken it.

**Comment by daniel65dj** [comment]
Well, that's a good consideration too. I think for students that are not familiar or struggle(d) with math at school should find this course beginner friendly - the lectures are of high visual quality, the assignments are not very hard and there are notes that accompany the lectures. And you can ask them directly if you have anything to ask.

**Comment by waciumawanjohi** [comment]
I am wary of replacing a course because another course teaches the prerequisite material. It seems to me that we should choose the best course for calculus and if precalculus is needed by a student, point them to the best resource for precalculus.

I am interested in hearing more about a direct comparison of current MIT recommendation and the new University of Sydney course. One approach might be to take a particular topic and analyze how each course teaches the material. You can see such an approach [here](https://github.com/ossu/computer-science/issues/465#issuecomment-386486394) comparing Ximera and Khan Academy's approach to the Squeeze Theorem.

There is a separate question: should OSSU make a recommendation for learning/reviewing precalculus? I'll note that OSSU focuses on the undergraduate curriculum and precalculus is generally within secondary school expectations.

There have been previous discussions of math prerequisites and what to recommend: https://github.com/ossu/computer-science/issues/438. The [prerequisite section](https://github.com/ossu/computer-science#prerequisites) links to Khan Academy when mentioning math. [Extras/Courses](https://github.com/ossu/computer-science/blob/master/extras/courses.md) links to a precalculus course that was rated one of the best new courses developed in 2019. hanjiexi's [v9 draft](https://github.com/ossu/computer-science/wiki/Foundations-draft) recommends a different course.

**Comment by daniel65dj** [comment]
I am also interested in a comparison between the MIT and Uni of Sydney courses.
I'll probably do the MIT courses next after I finish the Intro to Calculus and after that I'll share my thoughts.

**Comment by Alaharon123** [comment]
I leave it to @daniel65dj to talk about the quality of the course in comparison to MIT's, but I'll take a gander at why I think it's a better starting point for OSSU than MIT's from the perspective of just where the curriculum should start.

Who is OSSU for? OSSU is for everyone; it strives to be one size fits all (at least I think that's a goal). Starting with Introduction to Calculus is the perfect way to be one size fits all. Do you need some precalculus? Here you go. Are you cool with precalculus? Alright, skip the first two fifths of the course. Are you not sure? Skip the videos and text and just do the quizzes and if you're in over your head, go back and actually learn it.

The links you've given are mostly uneasy about MIT's courses being the recommended ones, albeit because of length, which can be remedied by switching to the OCW Scholar course instead. People seem to like the idea of using UPenn's coursera course instead, but are unsure what would be the prerequisite because this course didn't exist then to serve as a catchall for whatever knowledge you need. UPenn's course is an adaptation of their on-campus Math 104E, which had a prerequisite of Math 103 (unless there was a Math 103E then that was required). If we look at [the syllabus for Math 103](https://www.math.upenn.edu/sites/www.math.upenn.edu/files/math103syllabus19.pdf), it's very similar to Sydney's [Introduction to Calculus](https://www.coursera.org/learn/introduction-to-calculus).

To summarize, I don't think that OSSU should recommend a dedicated precalculus course for the reasons you mentioned. However, I think if this is indeed a great course, it would be a great first math course for OSSU for the flexibility it provides in working for all types of people: those who need to learn precalculus, those who need a refresher, and even those who don't need it at all. Additionally, it is worth considering switching out MIT's Calculus 1B and 1C with UPenn's Math 104E Calculus: Single Variable sequence.

**Comment by daniel65dj** [comment]
One technical question I have with Upenn courses: There are a total of five courses: four of them are parts of a one big course but there is a fifth one just named Single variable calculus with no part number on it.Is it a separate course or a part of the other four ?

**Comment by Alaharon123** [comment]
It should be called Calculus: Single Variable Part 5 - Discretization or Discrete Calculus. I don't know why it's missing the subtitle.

**Comment by daniel65dj** [comment]
After digging around here are all five parts listed: 
https://www.classcentral.com/course/calcsing-331

**Comment by waciumawanjohi** [comment]
I've left this open in order to invite further comment or support, but none has been offered since April.

I expressed above my wariness to replace a course simply because it covers more of the topic's prerequisites. It's certainly true that students come to OSSU needing to review math topics. Indeed, students regularly comment in gitter about their need to review not only precalculus, but also high school algebra. It does not seem that the proper approach is to suggest courses that blur the lines of their topic.
